distsys/archetype_resources/tcp_mailboxes.go

```diff
@@ -0,0 +1,331 @@
+package archetype_resources
+
+import (
+	"encoding/gob"
+	"fmt"
+	"github.com/UBC-NSS/pgo/distsys"
+	"log"
+	"net"
+	"time"
+)
+
+// Mailboxes as Archetype Resource
+// -------------------------------
+
+const (
+	tcpNetworkBegin = iota
+	tcpNetworkValue
+	tcpNetworkPreCommit
+	tcpNetworkCommit
+)
+
+type TCPMailboxKind int
+
+const (
+	TCPMailboxesLocal = iota
+	TCPMailboxesRemote
+)
+
+func TCPMailboxesArchetypeResourceMaker(addressMappingFn func(distsys.TLAValue) (TCPMailboxKind, string)) distsys.ArchetypeResourceMaker {
+	return IncrementalArchetypeMapResourceMaker(func(index distsys.TLAValue) distsys.ArchetypeResourceMaker {
+		tpe, addr := addressMappingFn(index)
+		switch tpe {
+		case TCPMailboxesLocal:
+			return tcpMailboxesLocalArchetypeResourceMaker(addr)
+		case TCPMailboxesRemote:
+			return tcpMailboxesRemoteArchetypeResourceMaker(addr)
+		default:
+			panic(fmt.Errorf("invalid TCP mailbox type %d for address %s: expected local or remote, which are %d or %d", tpe, addr, TCPMailboxesLocal, TCPMailboxesRemote))
+		}
+	})
+}
+
+type tcpMailboxesLocalArchetypeResource struct {
+	distsys.ArchetypeResourceLeafMixin
+	msgChannel chan distsys.TLAValue
+
+	readBacklog     []distsys.TLAValue
+	readsInProgress []distsys.TLAValue
+}
+
+var _ distsys.ArchetypeResource = &tcpMailboxesLocalArchetypeResource{}
+
+func tcpMailboxesLocalArchetypeResourceMaker(listenAddr string) distsys.ArchetypeResourceMaker {
+	return distsys.ArchetypeResourceMakerFn(func() distsys.ArchetypeResource {
+		msgChannel := make(chan distsys.TLAValue, 100)
```

Comment:
It's good to define constant variables for this kind of configurations (I mean size of the channel).

Reply:
Or, allow it to be configurable... yes, this would need fixing eventually one way or the other.
Question is how / is it worth the effort right now.

Reply:
I think making constants out of these numbers is easy and useful.